client/client.cpp

```diff
@@ -331,10 +331,10 @@ int open_can_socket(const char *can_interface_name, const struct can_filter *p_f
         error("ERROR setting loopback", errno);
     }
 
-    // Enable 1s timeout, so the thread is not blocking
+    // Enable 1ms timeout, so the thread is not blocking
     struct timeval tv;
-    tv.tv_sec = 1;
-    tv.tv_usec = 0;
+    tv.tv_sec = 0;
+    tv.tv_usec = 1000;
```

Comment:
How is this time selected? Compared to the previous 1s timeout, does it solve the issue completely or just reduce the chance of it happening?

Reply:
Also note that the timeout is not exact despite it seeming to be so with the units of usecs. The granularity varies for different installations and is usually on the the order of a few ms.

Reply:
The actual fix the removal of `continue` on line 444 of the original file. This would send the trapped messages after 1 second (or upon the next CAN transmission if that occurs sooner). One second is still a long latency, so it needs to be reduced to something smaller when there is pending data.